lib/wicked_pdf/option_parser.rb

```diff
@@ -61,7 +61,7 @@ def parse_header_footer(options)
           r += make_options(opt_hf, [:line], hf.to_s, :boolean)
           if options[hf] && options[hf][:content]
             @hf_tempfiles = [] unless defined?(@hf_tempfiles)
-            @hf_tempfiles.push(tf = WickedPdf::Tempfile.new("wicked_#{hf}_pdf.html"))
+            @hf_tempfiles.push(tf = File.new(Dir::Tmpname.create(["wicked_#{hf}_pdf", '.html']) {}, 'w'))
```

Comment:
Is there a reason `WickedPdf::Tempfile` doesn't work here? I remember adding that because an earlier version of Ruby didn't allow tempfiles to be created with an extension, which I know is no longer the case, but don't remember when that changed.

Reply:
@unixmonkey `WickedPdf::Tempfile` still can created the temp file. But the files is deleted before the pdf file generated.